src/core/server/integration_tests/saved_objects/migrations/zdt_2/v2_to_zdt_switch.test.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Path from 'path';
+import fs from 'fs/promises';
+import { range } from 'lodash';
+import { type TestElasticsearchUtils } from '@kbn/core-test-helpers-kbn-server';
+import { SavedObjectsBulkCreateObject } from '@kbn/core-saved-objects-api-server';
+import { IndexMappingMeta } from '@kbn/core-saved-objects-base-server-internal';
+import '../jest_matchers';
+import {
+  getKibanaMigratorTestKit,
+  startElasticsearch,
+  currentVersion,
+} from '../kibana_migrator_test_kit';
+import { delay, parseLogFile } from '../test_utils';
+import { getBaseMigratorParams, getSampleAType } from '../fixtures/zdt_base.fixtures';
+
+export const logFilePath = Path.join(__dirname, 'v2_to_zdt_switch.test.log');
+
+const v2IndexName = `.kibana_${currentVersion}_001`;
+
+describe('ZDT upgrades - switching from v2 algorithm', () => {
```

Comment:
Integration tests asserting the behavior (both for v2 compat and v2 incompat modes)